### PR TITLE
Remove lager

### PR DESCRIPTION
**Pull request by chrzaszcz**
The goal is to replace `lager` with Erlang `logger` in order to remove the `lager` dependency from [MongooseIM](https://github.com/esl/mongooseim/), and integrate the output with the MongooseIM logs.

It is not required to maintain all functionality that was present before, but only to keep the library functional enough to be used in the `mod_jingle_sip` module in MongooseIM. The changes are verified by the corresponding [PR to MongooseIM](https://github.com/esl/MongooseIM/pull/4393).

Additionally:
- Make `rebar3 compile` work on Erlang 27.
- Fix use of removed `erlang:get_stacktrace/0`.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 {erl_opts, [
-	debug_info,
-	fail_on_warning,
-    {parse_transform, lager_transform}
+	debug_info
 ]}.
 
 {deps, [
-        {nklib, ".*", {git, "https://github.com/michalwski/nklib.git", {ref, "006c1b7"}}},
+        {nklib, ".*", {git, "https://github.com/esl/nklib.git", {branch, "remove-lager"}}},
```
Comment (JanuszJakubiec) [inline rebar.config]: Maybe change the branch name to "mongooseim"?

File: include/nkpacket.hrl [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 -define(
     DO_LOG(Level, Domain, Text, Opts),
-        lager:Level([{domain, Domain}], "~p "++Text, [Domain|Opts])).
+        logger:Level("~p "++Text, [Domain|Opts])).
```
Comment (JanuszJakubiec) [inline include/nkpacket.hrl:36]: Is the “Level” function present in logger?

**Review by JanuszJakubiec — APPROVED** [review]
Looks good 👍🏻. I've added minor comment.

File: rebar.config (hunk 1 above)
Comment (chrzaszcz) [reply rebar.config]: Yes of course 🙂  I planned that, but I forgot to mark the PR as draft indeed.

**Review by JanuszJakubiec — APPROVED** [review]
Thanks for the change 👍🏻

**Pull request merged**